### PR TITLE
[CAPI] Expose layer enums

### PR DESCRIPTION
**Pull request by zhoonit**
- [CAPI] Expose layer enums

```
This patch contains layer enums to be exposed

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

File: api/nntrainer-api-common.h [hunk 1]
```diff
@@ -37,7 +37,29 @@ typedef enum {
...
+  ML_TRAIN_LAYER_TYPE_PREPROCESS_TRANSLATE, /**< Preprocess translate Layer
+                                               (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_PREPROCESS_L2NORM, /**< Preprocess L2Normalization Layer
+                                            (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_EMBEDDING,         /** Embedding Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_RNN,               /** RNN Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_LSTM,              /** LSTM Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_SPLIT,             /** Splite Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_GRU,               /** GRU Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_TIME_DIST, /** Time Distributed Layer (Since 6.5) */
```
Comment (zhoonit) [inline api/nntrainer-api-common.h:50]: Could someone else confirm this should be exposed as well?

Comment (kparichay) [reply api/nntrainer-api-common.h:50]: i this can be kept internal as for every layer, distribute can be set as a property.
@jijoongmoon can confirm this.

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: api/nntrainer-api-common.h (hunk 1 above)
Comment (jijoongmoon) [reply api/nntrainer-api-common.h]: Right. It can be kept internal. It is viewed as a property.

File: api/nntrainer-api-common.h [hunk 2]
```diff
@@ -37,7 +37,28 @@ typedef enum {
...
+  ML_TRAIN_LAYER_TYPE_EMBEDDING,           /** Embedding Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_RNN,                 /** RNN Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_LSTM,                /** LSTM Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_SPLIT,               /** Splite Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_GRU,                 /** GRU Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_PERMUTE,             /** Permute layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_DROPOUT,             /** DropOut Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_BACKBONE_NNSTREAMER, /**< Backbone using NNStreamer
+                                              (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_CENTROID_KNN, /**< Centroid KNN Layer (Since 6.5) */
```
Comment (kparichay) [inline api/nntrainer-api-common.h:54]: is this finalized?

Comment (zhoonit) [reply api/nntrainer-api-common.h:54]: I think so seeing that https://github.com/nnstreamer/nntrainer/issues/1421

Comment (kparichay) [reply api/nntrainer-api-common.h:54]: alright.

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**